kotsadm/pkg/dex/kotsadm.go

```diff
@@ -174,16 +184,34 @@ func stateSecretResource(secretName, state string) *corev1.Secret {
 			Labels: kotsadmtypes.GetKotsadmLabels(identity.DexAdditionalLabels),
 		},
 		Data: map[string][]byte{
-			state: []byte(state),
+			state: []byte(time.Now().UTC().Format(time.RFC3339)),
 		},
 	}
 }
 
 func updateStateSecret(existingSecret, desiredSecret *corev1.Secret) *corev1.Secret {
 	existingSecret.Data = mergeMaps(existingSecret.Data, desiredSecret.Data)
+	expireOldDexStates(existingSecret)
 	return existingSecret
 }
 
+func expireOldDexStates(secret *corev1.Secret) error {
+	for s, t := range secret.Data {
+		stateTime, err := time.Parse(time.RFC3339, string(t))
+		if err != nil {
+			return errors.Wrap(err, "failed to parse state time")
+		}
+
+		ttlTime := time.Now().Add(-5 * time.Minute)
```

Comment:
why 5 minutes? maybe you should make this something really long like an hour or 24 hours and let the oidc protocol enforce expiry if that is a thing?